Tools/CISupport/ews-build/results_db.py

```diff
@@ -48,25 +50,46 @@ class ResultsDatabase(object):
     @classmethod
     def get_results_summary(cls, test, commit=None, configuration=None):
         params = dict()
+        logs = ''
+        if not test:
+            return ({}, 'Test name not provided')
+        if not configuration:
+            configuration = {}
         for key, value in configuration.items():
             if key not in cls.CONFIGURATION_KEYS:
-                sys.stderr.write(f"'{key}' is not a valid configuration key\n")
+                logs += f"'{key}' is not a valid configuration key\n"
             params[key] = value
         if commit:
             params['ref'] = commit
         response = requests.get(f'{cls.HOSTNAME}/api/results-summary/{cls.SUITE}/{test}', params=params)
         if response.status_code != 200:
-            sys.stderr.write(f"Failed to query results summary with status code '{response.status_code}'\n")
-            return None
+            logs += f'Failed to query results summary with status code {response.status_code}\n'
+            return ({}, logs)
         try:
-            return response.json()
+            return (response.json(), logs)
         except json.decoder.JSONDecodeError:
-            sys.stderr.write('Non-json response from results summary query\n')
-            return None
+            logs += 'Non-json response from results summary query\n'
+            return ({}, logs)
+
+    @classmethod
+    def is_test_flaky_or_pre_existing(cls, test, commit=None, configuration=None):
+        data, logs = cls.get_results_summary(test, commit, configuration)
+        pass_rate = data.get('pass', 100) + data.get('warning', 0)
+        is_existing_failure = (pass_rate <= cls.PERECENT_SUCCESS_RATE_FOR_PRE_EXISTING_FAILURE)
+        is_flaky_failure = (pass_rate <= cls.PERECENT_SUCCESS_RATE_FOR_FLAKY_FAILURE) and not is_existing_failure
+        output = {
+            'is_existing_failure': is_existing_failure,
```

Comment:
I'm a bit unclear what benefit we get by trying to distinguish `flakey` and `existing` failures, because to EWS, they look very similar.

Consider 2 examples:
```
                    V
p-p-p-f-f-f-f-f-f-f-f-f-f
```

```
                    V
p-p-f-p-p-f-f-p-f-p-f-p-f
```
The former is clearly a recently added regression, the later is a flakey failure where we happen to be testing with a commit that failed. Both will have pretty similar rates, at least according the API. I don't think that EWS should differentiate between the two, there isn't a reliable way to automatically do that.

Reply:
Removed the concept of flaky when consulting results database in updated PR.